### PR TITLE
Astro 1829 aos los hide date time labels

### PR DESCRIPTION
**Pull request by Full-lifey**
## Brief Description

First, allow for aos and los properties to use either unix time or iso string.
Second, add hideLabels property to hide labels

## JIRA Link

[ASTRO-1829](https://rocketcom.atlassian.net/browse/ASTRO-1829?atlOrigin=eyJpIjoiYWZmYTA3MDMyODdmNDIwZmIyNDQ1YTg1MWUxMDJhYTQiLCJwIjoiaiJ9
[https://rocketcom.atlassian.net/browse/ASTRO-1844](https://rocketcom.atlassian.net/browse/ASTRO-1844?atlOrigin=eyJpIjoiMTRjZWVlYjAyYmY5NGE0Mzk2Y2E3YWI4OWE3NjQ5MTciLCJwIjoiaiJ9)

## Issues and Limitations

I decided to hide AOS and LOS labels as well. It was only specified to hide the date and time labels. However, I made the judgement call that if a user is going to hide the other labels they will hide those as well. I also thought it looked odd with some labels.

## Types of changes

-   [x] Bug fix
-   [x] New feature
-   [ ] Breaking change

## Checklist

-   [ ] This PR adds or removes a Storybook story.
-   [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/components/rux-clock/rux-clock.tsx [hunk 1]
```diff
@@ -14,17 +14,27 @@ export class RuxClock {
     private _timezone: string = 'UTC'
     private dayOfYear!: number
     private tzFormat: string = 'z'
+    private convertedAos?: string
+    private convertedLos?: string
```
Comment (Full-lifey) [inline src/components/rux-clock/rux-clock.tsx:18]: Added to internal variables so the method wouldn't run on every render (1s).

It doesn't seem computationally heavy to run the method every render but it felt like a waste to me so I chose to do it this way after the first iteration.

File: src/components/rux-clock/rux-clock.tsx [hunk 2]
```diff
@@ -108,23 +141,23 @@ export class RuxClock {
     render() {
         return (
             <Host>
-                {!this.hideDate ? (
+                {!this.hideDate && (
```
Comment (Full-lifey) [inline src/components/rux-clock/rux-clock.tsx:144]: Updated the formatting to simplify the syntax.

**Review by Full-lifey — COMMENTED** [review]
I'm curious about input around storing the AOS/LOS computed value. The big con is it requires more code and is therefore harder to read.

I could just call the method in the render but since we know the clock renders every second this extra computation felt like the bigger con to me.

**Review by markacianfrani — APPROVED** [review]

**Comment by amalyah** [comment]
When was small clock added to Storybook? I thought that PR got paused?

**Comment by markacianfrani** [comment]
> When was small clock added to Storybook? I thought that PR got paused?

That was part of the initial lit port

**Comment by amalyah** [comment]
Hide labels boolean in the UI is not doing anything in the sample for me.

**Comment by amalyah** [comment]
> > When was small clock added to Storybook? I thought that PR got paused?
> 
> That was part of the initial lit port

Okay. I wasn't aware that that existed before. Created a ticket for design: https://rocketcom.atlassian.net/browse/ASTRO-2054

**Review by amalyah — CHANGES_REQUESTED** [review]
Are the booleans for hide labels supposed to work in this PR?

**Comment by Full-lifey** [comment]
> Are the booleans for hide labels supposed to work in this PR?

Yes they are. It worked on the component but the storybook controls were not hooked up properly. 

This was a great catch and it has been resolved.

**Pull request merged**